src/nba/nbaStatSet.ts

```diff
@@ -0,0 +1,15 @@
+import { Stat, StatProduction, StatProductionType } from "../types/FantasyStats.js";
+
+export class NBAStatSet {
+    productionType: StatProductionType
+    stats: Map<string, Stat>
+
+    constructor(statProductionType: StatProductionType, statProduction: StatProduction) {
+        this.productionType = statProductionType;
+        this.stats = new Map<string, Stat>();
+
+        for (const [statName, statValue] of Object.entries(statProduction)) {
+            this.stats.set(statName, {name: statName, value: statValue, enabled: true})
```

Comment:
not sure what the statNames restrictions are, but might want some validation to ensure that the values are valid.

Reply:
Yeah good point - it'd be hard to score negative points for example. Will add some validation.